datafusion/core/src/execution/context.rs

```diff
@@ -1565,7 +1569,8 @@ impl SessionState {
             )));
         }
         physical_optimizers.push(Arc::new(Repartition::new()));
-        physical_optimizers.push(Arc::new(AddCoalescePartitionsExec::new()));
+        physical_optimizers.push(Arc::new(BasicEnforcement::new()));
+        // physical_optimizers.push(Arc::new(AddCoalescePartitionsExec::new()));
```

Comment:
why remove it?

Reply:
The new rule BasicEnforcement had already covered the work of AddCoalescePartitionsExec.

Reply:
Sure, I will do it in the following PR.